packages/next-server/lib/router/router.ts

```diff
@@ -67,6 +67,7 @@ export default class Router implements BaseRouter {
     this.asPath = as
     this.subscriptions = new Set()
     this.componentLoadCancel = null
+    this.parseParams()
```

Comment:
What is this doing?

Reply:
It's parsing the params, when on route `/$post/$commnet` it takes `/post-1/cmnt-1` and adds `{ post: 'post-1', comment: 'cmnt-1' }` to the query object.

Reply:
I think there's a better way to handle this

Reply:
Updated